### PR TITLE
consolidate images used for integration tests

### PR DESCRIPTION
**Pull request by sohankunkerkar**
Fixes https://issues.redhat.com/browse/OCPNODE-1535

#### What type of PR is this?

/kind ci
/kind cleanup

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: test/network.bats [hunk 1]
```diff
@@ -92,13 +87,11 @@ function teardown() {
 		"$TESTDATA"/sandbox_config.json > "$pod_config"
 
 	ctr_config="$TESTDIR"/container_config.json
-	jq '	  .image.image = "quay.io/crio/busybox:latest"
+	jq '	  .image.image = "quay.io/sohankunkerkar/fedora-crio-ci:minimal"
```
Comment (haircommander) [inline test/network.bats:90]: if it's all the same image, do we need to replace the image field here?

File: test/metrics.bats [hunk 2]
```diff
@@ -96,7 +96,7 @@ function teardown() {
 	PORT=$(free_port)
 	CONTAINER_ENABLE_METRICS=true CONTAINER_METRICS_PORT=$PORT start_crio
 
-	jq '.image.image = "quay.io/crio/oom"
+	jq '.image.image = "quay.io/sohankunkerkar/fedora-crio-ci:minimal"
```
Comment (haircommander) [inline test/metrics.bats:99]: same here--I think this part can be dropped

File: test/nri/runtime.go [hunk 3]
```diff
@@ -13,7 +13,7 @@ import (
 )
 
 const (
-	busyboxImage   = "quay.io/crio/busybox:latest"
+	busyboxImage   = "quay.io/sohankunkerkar/fedora-crio-ci:minimal"
```
Comment (haircommander) [inline test/nri/runtime.go:16]: nit: it's no longer busybox, maybe just call it `ciImage`?

File: test/testdata/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+ARG SU_EXEC_VERSION=0.2
+ARG SU_EXEC_URL="https://github.com/ncopa/su-exec/archive/v${SU_EXEC_VERSION}.tar.gz"
+RUN curl -sL "${SU_EXEC_URL}" | tar -C /tmp -zxf - \ 
+&& make -C "/tmp/su-exec-${SU_EXEC_VERSION}" \
+&& cp "/tmp/su-exec-${SU_EXEC_VERSION}/su-exec" /usr/bin \
+&& rm -fr "/tmp/su-exec-${SU_EXEC_VERSION}"
+
+ENV REDIS_VERSION 6.0.18
+ENV REDIS_DOWNLOAD_URL http://download.redis.io/releases/redis-6.0.18.tar.gz
+ENV REDIS_DOWNLOAD_SHA d7b4f2a97fcab96727284092b0a4aa854af47d570803fa0e7a0345359743836e
```
Comment (haircommander) [inline test/testdata/Dockerfile:34]: can you add a comment about how this is for redis image compatibility

**Comment by littlejawa** [comment]
I see the kata-jenkins CI fail systematically on :
- image volume ignore
- image volume user mkdir

The other tests you modified are passing with kata, and those two tests seem to pass in the other integration run (using runc), even if others are failing.

Sorry if you were already looking into it, just wanted to make sure you're aware.

As this PR is still WIP I didn't dig further, but if you think it's a kata-specific issue, I can help troubleshoot it.

**Comment by sohankunkerkar** [comment]
@cri-o/cri-o-maintainers PTAL

File: test/crio-wipe.bats [hunk 5]
```diff
@@ -124,7 +124,7 @@ function start_crio_with_stopped_pod() {
 	CONTAINER_INTERNAL_WIPE=false start_crio_with_stopped_pod
 	stop_crio_no_clean
 
-	run_podman_with_args run --name test -d quay.io/crio/busybox:latest top
+	run_podman_with_args run --name test -d quay.io/sohankunkerkar/fedora-crio-ci:minimal top
```
Comment (mrunalp) [inline test/crio-wipe.bats:127]: We should put this under quay.io/crio.

Comment (jrvaldes) [reply test/crio-wipe.bats:127]: +1

Comment (sohankunkerkar) [reply test/crio-wipe.bats:127]: Yeah, I don't have access to push that image under `quay.io/crio`

Comment (sohankunkerkar) [reply test/crio-wipe.bats:127]: @mrunalp is it okay to use a `fedora-minimal` image instead of a `ubi9-minimal` ?

File: test/pod.bats [hunk 6]
```diff
@@ -340,5 +340,6 @@ function teardown() {
 		"$TESTDATA"/container_sleep.json > "$etc_perm_config"
 	ctr_id=$(crictl create "$pod_id" "$etc_perm_config" "$TESTDATA"/sandbox_config.json)
 	output=$(crictl exec --sync "$ctr_id" ls -ld /etc)
+	echo "$output"
```
Comment (haircommander) [inline test/pod.bats:343]: leftover?

Comment (sohankunkerkar) [reply test/pod.bats]: Fixed

**Comment by haircommander** [comment]
good work here!

/approve

LGTM, @cri-o/cri-o-maintainers PTAL

**Review by mrunalp — APPROVED** [review]
/lgtm

**Pull request merged**